src/docfx/lib/HtmlUtility.cs

```diff
@@ -17,6 +17,87 @@ internal static class HtmlUtility
     {
         private static readonly string[] s_allowedStyles = new[] { "text-align: right;", "text-align: left;", "text-align: center;" };
 
+        // todo remove `class` & `hidden` after we can know it is is generated by ourself
+        private static readonly string[] s_globalAllowedAttributes = new[] { "id", "itemid", "itemprop", "itemref", "itemscope", "itemtype", "part", "slot", "spellcheck", "title", "role", "class", "hidden" };
+
+        // ref https://developer.mozilla.org/en-US/docs/Web/HTML/Element
+        private static readonly Dictionary<string, string[]> s_allowedTagAtrributeMap = new Dictionary<string, string>()
```

Comment:
Use `HashSet<string>` to speed up `Contains` test. Consider create a `string->string[]` dictionary directly over split strings at runtime. Could use `null` to represent `""`.

```suggestion
        private static readonly Dictionary<string, HashSet<string>?> s_allowedTagAtrributeMap = new Dictionary<string, string>()
```